### ISSUE TITLE
Wrong database quotes (MySQL)

### ISSUE THREAD

**Issue by szigya** [yiisoft/yii2#10344]
This is a part of my query:

```sql
    FLOOR( SUM ( IF(leaf_wetness>6, 1, 0 )) / 1440 ) AS leaf_sum
```
The executed part is this:
```sql
    FLOOR( SUM ( IF(leaf_wetness>6, `1`, `0 )) / 1440 )` AS `leaf_sum`
```

**Comment by rugabarbo** [comment]
How are you do it? Give us your PHP code.

**Comment by szigya** [comment]
The code mentioned above is an item in `$selectParts`.

```php
    $data = (new Query())
                ->select(implode(', ', $selectParts))
                ->from('{{%table}}')
                ->where(...)
                ->groupBy(implode(', ', $groupParts))
                ->all(Yii::$app->{'dbV4'});

**Comment by rugabarbo** [comment]
Use yii\db\Expression: http://www.yiiframework.com/doc-2.0/yii-db-expression.html
It will solve your problem.

**Comment by SilverFire** [comment]
`implode(', ', $selectParts)` it's a bad idea. Follow the guide: http://www.yiiframework.com/doc-2.0/guide-db-query-builder.html#select

**Comment by szigya** [comment]
I agree that it's bad, I removed that from the code, which solved my problem, so I can continue my work, but I still think that it shouldn't be closed or at least considered once again.
I mean, if I have just this expression in the select, it's still wrong. Is it the expected behavior?
Fixme, but I've seen in the code that there are filters for '{' and '(' which tells me that the framework is ready for expressions in strings and Expression class should be used when no escaping and quoting are needed.
tl;dr;
So this code is still doing the same. Is it the expected behaviour?
```php
$data = (new Query())
            ->select('SUM(IF(leaf_wetness>6,1,0)) AS leaf_sum')
            ->from('{{%table}}')
            ->where(..)
            ->all(Yii::$app->{'dbV4'});
```

Passing it as an array solves it, but for me it means that yii extracts the string in a bad way.

**Comment by SilverFire** [comment]
Yes, it's expected.

You may try:
`->select(new Expression('SUM(IF(leaf_wetness>6,1,0)) AS leaf_sum'))`

**Comment by szigya** [comment]
Okay, if it is the way it should work, I accept :). Thanks for the quick reply.
